网络年鉴大研究表明热门的前端框架依然是互联网的一小部分 (#8335)
xitu/gold-miner: article/2021/web-almanac-2020.md

> * 原文地址：[Web Almanac Mega Study Reveals That Popular Front-End Frameworks Are Still a Small Part of the Web](https://www.infoq.com/news/2021/03/web-almanac-2020)
> * 原文作者：[Bruno Couriol](https://www.infoq.com/profile/Bruno-Couriol/)
> * 译文出自：[掘金翻译计划](https://github.com/xitu/gold-miner)
> * 本文永久链接：[https://github.com/xitu/gold-miner/blob/master/article/2021/web-almanac-2020.md](https://github.com/xitu/gold-miner/blob/master/article/2021/web-almanac-2020.md)
> * 译者：[felixliao](https://github.com/felixliao)

> > * 校对者：

Reviewer: 增加校对者信息

> > * 校对者：

Author: ```suggestion
> * 校对者：[霜羽 Hoarfroster](https://github.com/PassionPenguin)、[Chor](https://github.com/Chorer)
```

> > - 原文地址：[Web Almanac Mega Study Reveals That Popular Front-End Frameworks Are Still a Small Part of the Web](https://www.infoq.com/news/2021/03/web-almanac-2020)
> > - 原文作者：Suresh Kumar
> > - 译文出自：[掘金翻译计划](https://github.com/xitu/gold-miner)
> > - 本文永久链接：[https://github.com/xitu/gold-miner/blob/master/article/2021/web-almanac-2020.md](https://github.com/xitu/gold-miner/blob/master/article/2021/web-almanac-2020.md)
> > - 译者：[felixliao](https://github.com/felixliao)
> > - 校对者：

Author: ```suggestion
> * 原文地址：[Web Almanac Mega Study Reveals That Popular Front-End Frameworks Are Still a Small Part of the Web](https://www.infoq.com/news/2021/03/web-almanac-2020)
> * 原文作者：Suresh Kumar
> * 译文出自：[掘金翻译计划](https://github.com/xitu/gold-miner)
> * 本文永久链接：[https://github.com/xitu/gold-miner/blob/master/article/2021/web-almanac-2020.md](https://github.com/xitu/gold-miner/blob/master/article/2021/web-almanac-2020.md)
> * 译者：[felixliao](https://github.com/felixliao)
> * 校对者：
```

# 网络年鉴大研究表明热门的前端框架依然是互联网的一小部分

> HTTP Archive 最近完成了 2020 网络年鉴，一份年度网络状况报告。这份报告从其四个部分（页面内容、用户体验、内容发表和分发），22 个章节和 600 页的内容中得出结论：jQuery 的使用率仍然高达 80%；CSS Houdini 很少被使用；2020 年网站发送的 JavaScript 大小中位数为 400 KB，比 2019 年多了 14%。除此以外，还有更多其他结论。

Author: ```suggestion
HTTP Archive 最近完成了 2020 网络年鉴，一份关于网络状况的年度报告。这份报告从四个部分（页面内容、用户体验、内容发表和分发）中的 22 个章节和 600 页中得出结论：jQuery 的使用率仍然高达 80%；CSS Houdini 很少被使用；2020 年网站发送的 JavaScript 大小中位数为 400 KB，比 2019 年多了 14%。除此以外，还有更多其他结论。
```

CSS 的章节强调，一些为丰富 CSS 标准的新功能或提案的采用可能会滞后：

> > 总的来看，我们观察到互联网上 CSS 的采用处在两个不同的层次。在博客和推文中我们更趋向于讨论最新最亮的东西，但是还是有成千上万的站点在使用十年前的代码。（……尽管）我们观察到许多新功能有引人注目的采用率 —— 甚至是那些今年才得到广泛支持的功能，例如 `min()` 和 `max()`，（……）总体上看，一个东西的酷炫程度和它的应用范围有多广还是呈反比；例如，先进的 Houdini 功能基本上不存在。（……）实践中大部分 CSS 的使用比较简单。

Author: 中文中表省略没有括号，直接用省略号，也许那样更好？

> > 总的来看，我们观察到互联网上 CSS 的采用处在两个不同的层次。在博客和推文中我们更趋向于讨论最新最亮的东西，但是还是有成千上万的站点在使用十年前的代码。（……尽管）我们观察到许多新功能有引人注目的采用率 —— 甚至是那些今年才得到广泛支持的功能，例如 `min()` 和 `max()`，（……）总体上看，一个东西的酷炫程度和它的应用范围有多广还是呈反比；例如，先进的 Houdini 功能基本上不存在。（……）实践中大部分 CSS 的使用比较简单。

Reviewer 2: > 中文中表省略没有括号，直接用省略号，也许那样更好？

没看出这里有省略的意思

> > 总的来看，我们观察到互联网上 CSS 的采用处在两个不同的层次。在博客和推文中我们更趋向于讨论最新最亮的东西，但是还是有成千上万的站点在使用十年前的代码。（……尽管）我们观察到许多新功能有引人注目的采用率 —— 甚至是那些今年才得到广泛支持的功能，例如 `min()` 和 `max()`，（……）总体上看，一个东西的酷炫程度和它的应用范围有多广还是呈反比；例如，先进的 Houdini 功能基本上不存在。（……）实践中大部分 CSS 的使用比较简单。

Reviewer 3: > > 中文中表省略没有括号，直接用省略号，也许那样更好？
> 
> 没看出这里有省略的意思

的确，我的另外一篇文也有大量的［...］，原文上也有，也不知道什么鬼👻

> > 总的来看，我们观察到互联网上 CSS 的采用处在两个不同的层次。在博客和推文中我们更趋向于讨论最新最亮的东西，但是还是有成千上万的站点在使用十年前的代码。（……尽管）我们观察到许多新功能有引人注目的采用率 —— 甚至是那些今年才得到广泛支持的功能，例如 `min()` 和 `max()`，（……）总体上看，一个东西的酷炫程度和它的应用范围有多广还是呈反比；例如，先进的 Houdini 功能基本上不存在。（……）实践中大部分 CSS 的使用比较简单。

Author: > > > 中文中表省略没有括号，直接用省略号，也许那样更好？
> > 
> > 没看出这里有省略的意思
> 
> 的确，我的另外一篇文也有大量的［...］，原文上也有，也不知道什么鬼👻

[] 是笔者补足的部分，比如如果只引用一个从句可能会补 [they] are xxx。里面加了省略号就是中间省略了一些片段。只是中文里没有中括号这个用法…

> > 总的来看，我们观察到互联网上 CSS 的采用处在两个不同的层次。在博客和推文中我们更趋向于讨论最新最亮的东西，但是还是有成千上万的站点在使用十年前的代码。（……尽管）我们观察到许多新功能有引人注目的采用率 —— 甚至是那些今年才得到广泛支持的功能，例如 `min()` 和 `max()`，（……）总体上看，一个东西的酷炫程度和它的应用范围有多广还是呈反比；例如，先进的 Houdini 功能基本上不存在。（……）实践中大部分 CSS 的使用比较简单。

Reviewer 2: > > > > 中文中表省略没有括号，直接用省略号，也许那样更好？
> > > 
> > > 
> > > 没看出这里有省略的意思
> > 
> > 
> > 的确，我的另外一篇文也有大量的［...］，原文上也有，也不知道什么鬼👻
> 
> [] 是笔者补足的部分，比如如果只引用一个从句可能会补 [they] are xxx。里面加了省略号就是中间省略了一些片段。只是中文里没有中括号这个用法…

嗯，你说的是对的，我忘了这段其实是引用报告的内容。去看了[报告原文的 CSS 部分](https://almanac.httparchive.org/en/2020/css)，发现中间确实省略了一些东西。但是按照汉语的习惯，引文内容有省略的时候，应该是不需要加括弧的，直接用省略号就行。所以我建议采用下面的翻译：
```suggestion
> 总的来看，我们观察到互联网上 CSS 的采用处在两个不同的层次。在博客和推文中我们更趋向于讨论最新、最亮眼的东西，但还是有成千上万的站点在使用十年前的代码。……尽管我们观察到许多新功能有很高的采用率 —— 甚至是那些今年才得到广泛支持的功能，例如 `min()` 和 `max()`，……但从总体上看，一个东西的酷炫程度和它的应用范围有多广还是呈反比；例如，先进的 Houdini 功能几乎像是不存在一样。……实践中大部分 CSS 的使用比较简单。
```

> CSS 的章节还指出 CSS 文件的中位大小在 60 KB 左右，基本上包含超过 400 条规则和 5000 条声明。大部分网站如今已在使用 [Flexbox 布局](https://css-tricks.com/snippets/css/a-guide-to-flexbox/)（2020 年为 60% 左右，2019 年为 40%）。CSS 网格布局（`display:grid`）正在被更多地使用（其使用率比起 2019 年翻了一番）但是依然是极其小众的（只有 5% 的网站使用）。过渡和动画使用非常广泛，超过 80% 的页面包含 `transition` 属性，超过 70% 包含 `animation` 属性。视觉效果（混合模式、图像滤镜、剪切路径）是非常热门的 CSS 功能：CSS 滤镜被使用于 80% 的页面；混合模式的占比和去年比翻倍，13% 的页面使用着它。

Author: ```suggestion
CSS 的章节还指出 CSS 文件的中位大小在 60 KB 左右，基本上包含超过 400 条规则和 5000 条声明。大部分网站如今已在使用 [Flexbox 布局](https://css-tricks.com/snippets/css/a-guide-to-flexbox/)（2020 年为 60% 左右，2019 年为 40%）。CSS 网格布局（`display:grid`）正在被更多地使用（其使用率比起 2019 年翻了一番）但是依然是极其小众的（只有 5% 的网站使用）。过渡和动画使用非常广泛，超过 80% 的页面包含 `transition` 属性，超过 70% 包含 `animation` 属性。视觉效果（混合模式、图像滤镜、剪切路径）是非常热门的 CSS 功能：CSS 滤镜被使用于 80% 的页面；混合模式的占比和去年比翻倍，13% 的页面在使用它。
```

相比之下，只有 2% 的网站使用任何 CSS-in-JS 的手段，尽管 CSS-in-JS 在开发者之间的讨论非常多。CSS Houdini 在网页中的使用也只有微乎其微的比例。该报告称：

> > CSS Painting API 是一个 Houdini 中被更广泛实现的规范，它允许开发者创建自定义 CSS 函数来返回 `<image>` 值，例如实现自定义渐变或图案。但是我们发现只有 12 个页面被发现使用了 `paint()`，且每个工作集名称 （`hexagon`、`ruler`、`lozenge`、`image-cross`、`grid`、`dashed-line`、`ripple`）在每个页面只出现了一次。由此看来，仅有的实践案例很可能是演示。Typed OM，另一个 Houdini 规范，允许了结构化数值的存取，而不是传统 CSS OM 中的字符串。比起其他 Houdini 规范，它有明显更高的采用率，尽管和整体比还是很低。它被应用于 9846 个桌面端站点（0.18%）和 6391 个移动端站点（0.1%）。

Author: ```suggestion
> CSS Painting API 是一个 Houdini 中被更为广泛实现的规范，它允许开发者创建自定义 CSS 函数来返回 `<image>` 值，例如实现自定义渐变或图案。但是我们发现只有 12 个页面在使用 `paint()`，且每个工作集名称 （`hexagon`、`ruler`、`lozenge`、`image-cross`、`grid`、`dashed-line`、`ripple`）在每个页面只出现了一次。由此看来，仅有的实践案例很可能是演示。Typed OM，另一个 Houdini 规范，允许了结构化数值的存取，而不是传统 CSS OM 中的字符串。比起其他 Houdini 规范，它有明显更高的采用率，尽管和整体比还是很低。它被应用于 9846 个桌面端站点（0.18%）和 6391 个移动端站点（0.1%）。
```

> 报告中的 JavaScript 章节称页面加载的 JavaScript 体积增长了 14%，其中中位页面会加载 400 KB 左右的 JavaScript。报告指出针对桌面端和移动端站点的页面有相近的 JavaScript 体积，但其网络和处理性能有很大差别；以及中位网站上有 37% 的 JavaScript 代码是没有被使用的。报告写道：

Author: ```suggestion
报告中的 JavaScript 章节称页面加载的 JavaScript 体积增长了约 14%，其中中位页面会加载 400 KB 左右的 JavaScript。报告指出针对桌面端和移动端的页面有相近的 JavaScript 体积，但其网络和处理性能有很大差别；以及中位网站上有 37% 的 JavaScript 代码是没有被使用的。报告写道：
```

> 报告中的 JavaScript 章节称页面加载的 JavaScript 体积增长了约 14%，统计的中位数表明会加载 400 KB 左右的 JavaScript。报告指出针对桌面端和移动端的页面有相近的 JavaScript 体积，但其网络和性能处理有很大差别；同时，中位网站上有 37% 的 JavaScript 代码是没有被使用的。报告写道：

Author: ```suggestion
报告中的 JavaScript 章节称页面加载的 JavaScript 体积增长了约 14%，统计的中位数表明会加载 400 KB 左右的 JavaScript。报告指出针对桌面端和移动端的页面有相近的 JavaScript 体积，但其网络和性能处理有很大差别；同时，网站上未使用的 JavaScript 代码的比例中位数达到了 37%。报告写道：
```

> > 取决于你的看法，你也许可以宽恕自己没有因为发送到桌面端和移动端代码间小小的区别感到懊恼 — 毕竟，30 KB 的中位差能怎么样呢，对吧？[...] 中位桌面站点花费浏览器主进程 891 毫秒来处理这些 JavaScript。然而，中位移动端站点花费 1897 毫秒 — 是桌面端花费时间的两倍多。这在底层的页面上更为糟糕。在 90 百分位，移动端页面要占据主进程 8921 毫秒来处理 JavaScript，相比之下桌面站点只需要 3838 毫秒。

Reviewer 2: ```suggestion
> 取决于你的看法，你也许可以不在乎发送到桌面端和移动端代码体积之间细微的区别 —— 毕竟，中位数差个 30 KB 又能怎么样呢，对吧？桌面端站点数据的中位数表明，其花费了 891 毫秒来去处理浏览器主进程上的 JavaScript。然而，移动端站点数据的中位数则是 1897 毫秒 —— 是桌面端花费时间的两倍多。这在底层的页面上更为糟糕。在数据的 90 百分位处，移动端站点要花费 8921 毫秒去处理主进程的 JavaScript，相比之下桌面端站点只需要 3838 毫秒。
```

这个中位数还是要翻译出来，说中位桌面站点的话可能没人知道是在指什么……

> > 取决于你的看法，你也许可以宽恕自己没有因为发送到桌面端和移动端的代码间小小的差距感到懊恼 — 毕竟，30 KB 的中位差能怎么样呢，对吧？[...] 中位桌面站点花费浏览器主进程 891 毫秒来处理这些 JavaScript。然而，中位移动端站点花费 1897 毫秒 — 桌面端花费时间的两倍多。这在底部的页面上更为糟糕。在 90 百分位，移动端页面要占据主进程 8921 毫秒来处理 JavaScript，相比之下桌面站点只需要 3838 毫秒。

Author: ```suggestion
> 取决于你的看法，你也许可以宽恕自己没有因为发送到桌面端和移动端代码间小小的区别感到懊恼 — 毕竟，30 KB 的中位差能怎么样呢，对吧？[...] 中位桌面站点花费浏览器主进程 891 毫秒来处理这些 JavaScript。然而，中位移动端站点花费 1897 毫秒 — 是桌面端花费时间的两倍多。这在底层的页面上更为糟糕。在 90 百分位，移动端页面要占据主进程 8921 毫秒来处理 JavaScript，相比之下桌面站点只需要 3838 毫秒。
```

> 好的一面是，[resource hints](https://www.infoq.com/news/2019/09/webexpo-2019-resource-hints-tips/) 可以抵消一些加载 JavaScript 的网络成本，它在将近 17% 的页面中被使用，其中大部分来自于 [`preload`](https://developer.mozilla.org/en-US/docs/Web/HTML/Preloading_content)。

Reviewer 2: ```suggestion
好的一面是，[resource hints](https://www.infoq.com/news/2019/09/webexpo-2019-resource-hints-tips/) 可以抵消一些加载 JavaScript 的网络成本，有将近 17% 的页面使用了这项技术，其中大部分来自于 [`preload`](https://developer.mozilla.org/en-US/docs/Web/HTML/Preloading_content)。
```

> 报告中关于第三方的章节分析了第三方脚本的存在、分布和影响，强调了管理它们的重要性：

Author: ```suggestion
报告中关于第三方的章节分析了第三方脚本的存在、分布和影响，并强调了管理它们的重要性：
```

> 好的一面是，[resource hints](https://www.infoq.com/news/2019/09/webexpo-2019-resource-hints-tips/) 可以抵消一些加载 JavaScript 的网络成本，它在将近 17% 的页面汇总被使用，其中大部分来自于 [`preload`](https://developer.mozilla.org/en-US/docs/Web/HTML/Preloading_content)。

Author: ```suggestion
好的一面是，[resource hints](https://www.infoq.com/news/2019/09/webexpo-2019-resource-hints-tips/) 可以抵消一些加载 JavaScript 的网络成本，它在将近 17% 的页面中被使用，其中大部分来自于 [`preload`](https://developer.mozilla.org/en-US/docs/Web/HTML/Preloading_content)。
```

> > 广告相关的请求似乎对 CPU 时间有负面影响。包含广告脚本的中位页面，比起那些不含脚本的，花费了最多三倍 CPU 时间。

Reviewer 2: ```suggestion
> 广告相关的请求似乎对 CPU 时间有负面影响。在中位数数据的比较上，包含广告脚本的页面花费的 CPU 时间，是那些不包含脚本的页面花费时间的三倍。
```

> > 广告相关的请求似乎对 CPU 时间有更大的影响。包含广告脚本的中位页面，比起那些不含脚本的，花费了最多三倍 CPU 时间。

Author: ```suggestion
> 广告相关的请求似乎对 CPU 时间有负面影响。包含广告脚本的中位页面，比起那些不含脚本的，花费了最多三倍 CPU 时间。
```

> 开发者们可以用四种语言[在线参考完整的报告](https://almanac.httparchive.org/en/2020/)。HTTP Archive 花费超过五个月来编写互联网年鉴，数百计的贡献者搜集了数百万网页的数据。HTTP Archive 每月评估数百万网页的构成，并将其 TB 级的元数据公开在 [BigQuery](https://httparchive.org/faq#how-do-i-use-bigquery-to-write-custom-queries-over-the-data) 上供分析使用。

Reviewer 2: ```suggestion
开发者们可以用四种语言[阅读完整的在线报告](https://almanac.httparchive.org/en/2020/)。HTTP Archive 耗时超过五个月来编写网络年鉴，其中更有数百计的贡献者搜集了数百万网页的数据。HTTP Archive 每月评估数百万网页的构成，并将其 TB 级的元数据公开在 [BigQuery](https://httparchive.org/faq#how-do-i-use-bigquery-to-write-custom-queries-over-the-data) 上供分析使用。
```

> 开发者们可以用四种语言[在线参考完整的报告](https://almanac.httparchive.org/en/2020/)。HTTP Archive 花费超过五个月来编写互联网年鉴，数百计的贡献者搜集了数百万网页的数据。HTTP Archive 每月评估数百万网页的组成，并将其数 TB 的元数据公开在 [BigQuery](https://httparchive.org/faq#how-do-i-use-bigquery-to-write-custom-queries-over-the-data) 上供分析使用。

Author: ```suggestion
开发者们可以用四种语言[在线参考完整的报告](https://almanac.httparchive.org/en/2020/)。HTTP Archive 花费超过五个月来编写互联网年鉴，数百计的贡献者搜集了数百万网页的数据。HTTP Archive 每月评估数百万网页的构成，并将其 TB 级的元数据公开在 [BigQuery](https://httparchive.org/faq#how-do-i-use-bigquery-to-write-custom-queries-over-the-data) 上供分析使用。
```

> > * 原文作者：Suresh Kumar

Reviewer 3: ```suggestion
> * 原文作者：[Bruno Couriol](https://www.infoq.com/profile/Bruno-Couriol/)
```

> * 译文出自：[掘金翻译计划](https://github.com/xitu/gold-miner)
> * 本文永久链接：[https://github.com/xitu/gold-miner/blob/master/article/2021/web-almanac-2020.md](https://github.com/xitu/gold-miner/blob/master/article/2021/web-almanac-2020.md)
> * 译者：[felixliao](https://github.com/felixliao)
> * 校对者：

# 网络年鉴大研究表明热门的前端框架依然是互联网的一小部分

> HTTP Archive 最近完成了 2020 网络年鉴，一份年度网络状况报告。这份报告从其四个部分（页面内容、用户体验、内容发表和分发），22 个章节和 600 页的内容中得出结论：jQuery 依旧占据着 80% 的互联网；CSS Houdini 很少被使用；2020 年网站发送的 JavaScript 大小中位数为 400 KB，比 2019 年多了 14%，还有更多。

Reviewer 3: ```suggestion
HTTP Archive 最近完成了 2020 网络年鉴，一份年度网络状况报告。这份报告从其四个部分（页面内容、用户体验、内容发表和分发），22 个章节和 600 页的内容中得出结论：jQuery 的使用率仍然高达 80%；CSS Houdini 很少被使用；2020 年网站发送的 JavaScript 大小中位数为 400 KB，比 2019 年多了 14%以及更多其他结论。
```

CSS 的章节强调，一些为丰富 CSS 标准的新功能或提案的采用可能会滞后：

> > 总的来看，我们观察到互联网上 CSS 的采用处在两个不同的档位。在博客和推文中，我们更趋向于讨论最新最亮的东西，但是， 还是有成千上万的站点在使用十年前的代码。[... 尽管] 我们观察到许多新功能有引人注目的采用率 —— 甚至是那些今年才得到广泛支持的功能，例如 `min()` 和 `max()`，[...] 总体上看，一个东西的酷炫程度和它的应用范围有多广还是呈反比；例如，先进的 Houdini 功能基本上不存在。[...] 实践中大部分 CSS 的使用比较简单。

Reviewer 3: ```suggestion
> 总的来看，我们观察到互联网上 CSS 的采用处在两个不同的档位。在博客和推文中，我们更趋向于讨论最新最亮的东西，但是，还是有成千上万的站点在使用十年前的代码。[... 尽管] 我们观察到许多新功能有引人注目的采用率 —— 甚至是那些今年才得到广泛支持的功能，例如 `min()` 和 `max()`，[...] 总体上看，一个东西的酷炫程度和它的应用范围有多广还是呈反比；例如，先进的 Houdini 功能基本上不存在。[...] 实践中大部分 CSS 的使用比较简单。
```
```suggestion
> 总的来看，我们观察到互联网上 CSS 的采用处在两个不同的档位。在博客和推文中，我们更趋向于讨论最新最亮的东西，但是， 还是有成千上万的站点在使用十年前的代码。[... 尽管] 我们观察到许多新功能有引人注目的采用率 —— 甚至是那些今年才得到广泛支持的功能，例如 `min()` 和 `max()`，[...] 总体上看，一个东西的酷炫程度和它的应用范围有多广还是呈反比；例如，先进的 Houdini 功能基本上不存在。[...] 实践中大部分 CSS 的使用比较简单。
```

> > 总的来看，我们观察到互联网上 CSS 的采用处在两个不同的档位。在博客和推文中，我们更趋向于讨论最新最亮的东西，但是， 还是有成千上万的站点在使用十年前的代码。[... 尽管] 我们观察到许多新功能有引人注目的采用率 —— 甚至是那些今年才得到广泛支持的功能，例如 `min()` 和 `max()`，[...] 总体上看，一个东西的酷炫程度和它的应用范围有多广还是呈反比；例如，先进的 Houdini 功能基本上不存在。[...] 实践中大部分 CSS 的使用比较简单。

Reviewer 3: ```suggestion
> 总的来看，我们观察到互联网上 CSS 的采用处在两个不同的层次。在博客和推文中我们更趋向于讨论最新最亮的东西，但是还是有成千上万的站点在使用十年前的代码。（……尽管）我们观察到许多新功能有引人注目的采用率 —— 甚至是那些今年才得到广泛支持的功能，例如 `min()` 和 `max()`，（……）总体上看，一个东西的酷炫程度和它的应用范围有多广还是呈反比；例如，先进的 Houdini 功能基本上不存在。（……）实践中大部分 CSS 的使用比较简单。
```

> CSS 的章节还指出 CSS 的中位大小在 60 KB 左右，包含超过 400 条规则和 5000 条声明。大部分网站如今已在使用 [Flexbox 布局](https://css-tricks.com/snippets/css/a-guide-to-flexbox/)（2020 年为 60% 左右，2019 年为 40%）。CSS 网格布局（`display:grid`）正在被更多地使用（其使用率比起 2019 年翻了一番）但是依然是极其小众的（只有 5% 的网站使用）。过渡和动画使用非常广泛，超过 80% 的页面包含 `transition` 属性，超过 70% 包含 `animation` 属性。视觉效果（混合模式、图像滤镜、剪切路径）是非常热门的 CSS 功能：CSS 滤镜被使用于 80% 的页面；混合模式的占比和去年比翻倍，达到了 13% 的页面。

Reviewer 3: ```suggestion
CSS 的章节还指出 CSS 文件的中位大小在 60 KB 左右，基本上包含超过 400 条规则和 5000 条声明。大部分网站如今已在使用 [Flexbox 布局](https://css-tricks.com/snippets/css/a-guide-to-flexbox/)（2020 年为 60% 左右，2019 年为 40%）。CSS 网格布局（`display:grid`）正在被更多地使用（其使用率比起 2019 年翻了一番）但是依然是极其小众的（只有 5% 的网站使用）。过渡和动画使用非常广泛，超过 80% 的页面包含 `transition` 属性，超过 70% 包含 `animation` 属性。视觉效果（混合模式、图像滤镜、剪切路径）是非常热门的 CSS 功能：CSS 滤镜被使用于 80% 的页面；混合模式的占比和去年比翻倍，13% 的页面使用着它。
```

相比之下，只有 2% 的网站使用任何 CSS-in-JS 的手段，尽管 CSS-in-JS 在开发者之间的讨论非常多。CSS Houdini 在网页中的使用也只有微乎其微的比例。该报告称：

> > CSS Painting API 是一个 Houdini 中被更广泛实现的规范，它允许开发者创建自定义 CSS 函数来返回 <image> 值，例如实现自定义渐变或图案。只有 12 个页面被发现使用了 paint()。每个工作集名称 （hexagon、ruler、lozenge、image-cross、grid、dashed-line、ripple）在每个页面只出现了一次，因此看来，仅有的实践案例很可能是演示。Typed OM，另一个 Houdini 规范，允许了结构化数值的存取，而不是传统 CSS OM 中的字符串。比起其他 Houdini 规范，它有明显更高的采用率，尽管和整体比还是很低。它被应用于 9846 个桌面端站点（0.18%）和 6391 个移动端站点（0.1%）。

Reviewer 3: ```suggestion
> CSS Painting API 是一个 Houdini 中被更广泛实现的规范，它允许开发者创建自定义 CSS 函数来返回 `<image>` 值，例如实现自定义渐变或图案。但是我们发现只有 12 个页面被发现使用了 `paint()`，且每个工作集名称 （`hexagon`、`ruler`、`lozenge`、`image-cross`、`grid`、`dashed-line`、`ripple`）在每个页面只出现了一次。由此看来，仅有的实践案例很可能是演示。Typed OM，另一个 Houdini 规范，允许了结构化数值的存取，而不是传统 CSS OM 中的字符串。比起其他 Houdini 规范，它有明显更高的采用率，尽管和整体比还是很低。它被应用于 9846 个桌面端站点（0.18%）和 6391 个移动端站点（0.1%）。
```

Reviewer 3: painting api 真的不好用……太麻烦

> 报告中的 JavaScript 章节称页面加载的 JavaScript 体积增长了 14%，其中中位页面加载了 400 KB 左右的 JavaScript。报告指出针对桌面端和移动端站点的页面有相近的 JavaScript 体积，但其网络和处理性能有很大差别；以及中位网站上有 37% 的 JavaScript 是没有被使用的。报告写道：

Reviewer 3: ```suggestion
报告中的 JavaScript 章节称页面加载的 JavaScript 体积增长了 14%，其中中位页面会加载 400 KB 左右的 JavaScript。报告指出针对桌面端和移动端站点的页面有相近的 JavaScript 体积，但其网络和处理性能有很大差别；以及中位网站上有 37% 的 JavaScript 代码是没有被使用的。报告写道：
```

> 取决于你的看法，你也许可以宽恕自己没有因为发送到桌面端和移动端的代码间小小的差距感到懊恼 — 毕竟，30 KB 的中位差能怎么样呢，对吧？[...] 中位桌面站点花费浏览器主进程 891 毫秒来处理这些 JavaScript。然而，中位移动端站点花费 1897 毫秒 — 桌面端花费时间的两倍多。这在底部的页面上更为糟糕。在 90 百分位，移动端页面要占据主进程 8921 毫秒来处理 JavaScript，相比之下桌面站点只需要 3838 毫秒。

好的一面是，[resource hints](https://www.infoq.com/news/2019/09/webexpo-2019-resource-hints-tips/) 可以抵消一些加载 JavaScript 的网络成本，它在将近 17% 的页面中被使用，其中大部分来自于 [`preload`](https://developer.mozilla.org/en-US/docs/Web/HTML/Preloading_content)。

报告中关于第三方的章节分析了第三方脚本的存在、分布和影响，强调了管理它们的重要性：

> 广告相关的请求似乎对 CPU 时间有更大的影响。包含广告脚本的中位页面，比起那些不含脚本的，花费了最多三倍 CPU 时间。

> 开发者们可以用四种语言 [在线参考完整的报告](https://almanac.httparchive.org/en/2020/)。HTTP Archive 花费超过五个月来编写互联网年鉴，数百计的贡献者搜集了数百万网页的数据。HTTP Archive 每月评估数百万网页的组成，并将其数 TB 的元数据公开在 [BigQuery](https://httparchive.org/faq#how-do-i-use-bigquery-to-write-custom-queries-over-the-data) 上供分析使用。

Reviewer 3: ```suggestion
开发者们可以用四种语言[在线参考完整的报告](https://almanac.httparchive.org/en/2020/)。HTTP Archive 花费超过五个月来编写互联网年鉴，数百计的贡献者搜集了数百万网页的数据。HTTP Archive 每月评估数百万网页的组成，并将其数 TB 的元数据公开在 [BigQuery](https://httparchive.org/faq#how-do-i-use-bigquery-to-write-custom-queries-over-the-data) 上供分析使用。
```
```suggestion
开发者们可以用四种语言 [在线参考完整的报告](https://almanac.httparchive.org/en/2020/)。HTTP Archive 花费超过五个月来编写互联网年鉴，数百计的贡献者搜集了数百万网页的数据。HTTP Archive 每月评估数百万网页的组成，并将其数 TB 的元数据公开在 [BigQuery](https://httparchive.org/faq#how-do-i-use-bigquery-to-write-custom-queries-over-the-data) 上供分析使用。
```
